app/src/utils.test.js

```diff
@@ -0,0 +1,40 @@
+import { canExecuteDeepLinking } from './utils';
+
+describe('utils', () => {
+  describe('canExecuteDeepLinking', () => {
+    it('Should return valid if url path is whitelisted and has required search params', () => {
+      const urlWithoutMessage =
+        'lisk://wallet?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=1&token=0200000000000000&recipientChain=02000000';
+      const urlWithMessage =
+        'lisk://wallet?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=1&token=0200000000000000&recipientChain=02000000&reference=test message';
+
+      expect(canExecuteDeepLinking(urlWithoutMessage)).toBeTruthy();
+      expect(canExecuteDeepLinking(urlWithMessage)).toBeTruthy();
+    });
+
+    it('Should return invalid if url path is not whitelisted and has required search params', () => {
+      const url =
+        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=1&token=0200000000000000&recipientChain=02000000';
+      expect(canExecuteDeepLinking(url)).toBeFalsy();
+    });
+
+    it('Should return invalid if url path is whitelisted and does not have required search params', () => {
+      const invalidExtraParamUrl =
+        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=1&token=0200000000000000&recipientChain=02000000&test=test';
+      const invalildAmountUrl =
+        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=test&token=0200000000000000&recipientChain=02000000';
+      const invalidChainIDUrl =
+        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=1&token=0200000000000000&recipientChain=02000';
+      const invalidTokenIDUrl =
+        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=1&token=00000000000&recipientChain=02000000';
+      const invalidAddressUrl =
+        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8&amount=1&token=0200000000000000&recipientChain=02000000';
+
+      expect(canExecuteDeepLinking(invalidExtraParamUrl)).toBeFalsy();
+      expect(canExecuteDeepLinking(invalildAmountUrl)).toBeFalsy();
+      expect(canExecuteDeepLinking(invalidChainIDUrl)).toBeFalsy();
+      expect(canExecuteDeepLinking(invalidTokenIDUrl)).toBeFalsy();
+      expect(canExecuteDeepLinking(invalidAddressUrl)).toBeFalsy();
```

Comment:
```suggestion
      const invalidDeepLinkURLs = [
        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=1&token=0200000000000000&recipientChain=02000000&test=test',
        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=test&token=0200000000000000&recipientChain=02000000',
        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=1&token=0200000000000000&recipientChain=02000',
        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8q2549oc&amount=1&token=00000000000&recipientChain=02000000',
        'lisk://test?modal=send&recipient=lskdb3ja5dhr25jhwm6onstawj33yk74g8&amount=1&token=0200000000000000&recipientChain=02000000'];

      test.each(invalidDeepLinkURLs)('should return true when isValidWCURI receives %s', (invalidURL) => {
          expect(canExecuteDeepLinking(invalidURL)).toBe(false);
      });
```

Reply:
This method looks clean though if done this way, it would be difficult for an external person taking a look at the code to tell the difference between the urls reason why I provided variable names to each to describe them. Though I can still do what you have proposed but some things like this

```
const invalidDeepLinkURLs = [
 invalidExtraParamUrl,
 invalildAmountUrl,
 invalidChainIDUrl,
 invalidTokenIDUrl,
 invalidAddressUrl
]
```